src/common/cmdparse.cc

```diff
@@ -14,13 +14,37 @@
 
 #include <cxxabi.h>
 #include "common/cmdparse.h"
+#include "common/Formatter.h"
 #include "include/str_list.h"
 #include "json_spirit/json_spirit.h"
 #include "common/debug.h"
 
 using namespace std;
 
 /**
+ * Given a cmddesc like "foo baz name=bar,type=CephString",
+ * return the prefix "foo baz".
+ */
+std::string cmddesc_get_prefix(const std::string &cmddesc)
+{
+  stringstream ss(cmddesc);
+  std::string word;
+  std::ostringstream result;
+  bool first = true;
+  while (std::getline(ss, word, ' ')) {
+    if (word.find_first_of(",=") == string::npos) {
```

Comment:
The parameter list is indeed space separated, so the rfind idea doesn't work.  Changed this to break when we see a ",=" match as suggested.